submissions/DmytryjK/task_js_dom/js/script.js

```diff
@@ -0,0 +1,186 @@
+
+document.addEventListener('DOMContentLoaded', startEventsForPage());
+
+function startEventsForPage() {
+    createElements();
+    currentTabButton('main__navigation-button_active', '.main__navigation-list');
+    addClass(mainContentBlock, 'main__content-active');
+    
+    function currentTabButton(tabSelector, parrentSelector) {
+        const parrentMenuWrap = document.querySelector(parrentSelector);
+
+        addClass(navigationButton, tabSelector);
+        parrentMenuWrap.addEventListener('click', e => {
+            let index = +e.target.getAttribute("data-number");
+            e.preventDefault();
+            
+            removeClass(navigationButton, tabSelector);
+            removeClass(mainContentBlock, 'main__content-active');
+
+            e.target.classList.add(tabSelector);
+
+            deleteInnerHtml();
+
+            createElements(index);
+            addClass(mainContentBlock, 'main__content-active', index);
+        });
+    };
+
+    function createElements (index = 0) {
+        mainContentBlock[index].innerHTML = `
+            <h2 class="main__content-title">${contentArray[index].title}</h2>
+            <p class="main__content-subtitle">${contentArray[index].subtitle}</p>
+            <img class="main__content-image" src="${contentArray[index].imgSrc}" alt="${contentArray[index].imgAlt}">
+            <p class="main__content-descr">${contentArray[index].paragraphsArray[0]}</p>
+            <p class="main__content-descr">${contentArray[index].paragraphsArray[1]}</p>
+        `;
+    };
+
+    function deleteInnerHtml() {
+        mainContentBlock.forEach(element => {
+            element.innerHTML = "";
+        });
+    };
+
+    function addClass(element, activeClass, index = 0) {
+        element[index].classList.add(activeClass);
+    };
+
+    function removeClass(element, activeClass) {
+        element.forEach(item => {
+            item.classList.remove(activeClass);
+        });
+    };
+
+    //Hamburger
+    const humbergerBtn = document.querySelector('.header__hamburger'),
+        adaptiveTabsMenu = document.querySelector('.adaptive'),
+        adaptiveNavTabsBtn = document.querySelectorAll('.adaptive__navigation-button'),
+        body = document.querySelector('body');
+
+    addClass(adaptiveNavTabsBtn, 'adaptive__navigation-button__active');    
+    openContentWithTab(adaptiveNavTabsBtn);  
+
+    humbergerBtn.addEventListener('click', e => {
+        e.preventDefault();
+        e.currentTarget.classList.toggle('header__hamburger-active');
+
+        body.classList.toggle('body__active');
+
+        toggleTabs(adaptiveTabsMenu, 'adaptive__active'); 
+    });
+
+    function toggleTabs(element, activeClass) {
+        element.classList.toggle(activeClass);
+    };
+
+    function openContentWithTab(navBtn) {
+        navBtn.forEach((button, index) => {
+            button.addEventListener('click', e => {
+                e.preventDefault();
```

Comment:
Any reason why you need `preventDefault` on click event on button?

Reply:
I checked it, and you are right, in this situation, it is not needed